src/search/landmarks/util.h

```diff
@@ -20,6 +21,8 @@ bool _possibly_reaches_lm(const OperatorProxy &op,
 
 OperatorProxy get_operator_or_axiom(const TaskProxy &task_proxy, int op_or_axiom_id);
 int get_operator_or_axiom_id(const OperatorProxy &op);
+
+void dump(const TaskProxy &task_proxy, const LandmarkGraph &graph);
```

Comment:
Shouldn't this be a const method?

Reply:
I'm fine with both but I thought we wanted to move in a direction of more functional code. The `dump` method only requires access to the public interface of `LandmarkGraph` and moving it to the util header makes the landmark graph code easier to understand. The code is not used currently, so completely removing it would also be an option but I thought it might be useful for debugging in the future.

Reply:
Sorry, I thought this was inside a class. If this is a free-standing function, I think we normally use `extern` explicitly.

Reply:
So to elaborate, my suggestion was to have this a const method instead of a non-const method, but a free function is better than either. Of course it should be clear that it isn't a method at all from the fact that a landmark graph reference is passed in, but I did not think about it too carefully. For a free function, I wonder if `dump` is a bit generic as a name, though. But I'm not 100% against it.

Reply:
I think we have others like this, so I'd leave it like this for now. I added the `extern` keywords for this and the other functions in this file.

Reply:
I had a little look at the rest of the code:
```
$ grep 'extern.*dump' **/*.{h,cc}
pdbs/utils.h:extern void dump_pattern_generation_statistics(
pdbs/utils.h:extern void dump_pattern_collection_generation_statistics(
task_utils/task_properties.h:extern void dump_pddl(const State &state);
task_utils/task_properties.h:extern void dump_fdr(const State &state);
task_utils/task_properties.h:extern void dump_goals(const GoalsProxy &goals);
task_utils/task_properties.h:extern void dump_task(const TaskProxy &task_proxy);
task_utils/variable_order_finder.h:extern void dump_variable_order_type(VariableOrderType variable_order_type);
```

Reply:
I must have misremembered this. I renamed the function.